### PR TITLE
Update to last 1.12 mappings

### PR DESCRIPTION
**Pull request by ichttt**
As stable 39 is the last 1.12 mapping version ever, this is a good opportunity to update the mappings. It would be best to merge other PRs targeting 1.12 first, so they don't need to update the mappings for their PR

### CODE REVIEW

**Comment by strikerrocker** [comment]
May as well update the default gradle wrapper packaged by forge?

File: build.gradle [hunk 1]
```diff
@@ -27,7 +27,7 @@ apply plugin: "net.minecraftforge.gradle.launch4j"
 minecraft.version = "1.12.2"
 
 minecraft {
-    mappings = 'snapshot_nodoc_20171003'
+    mappings = 'stable_39'
```
Comment (bs2609) [inline build.gradle]: This needs to be `stable_nodoc_39`

**Comment by bs2609** [comment]
There's also some overzealous find/replace going on here - the `Fluid` methods here aren't from vanilla and need to be kept for now to preserve compatibility.

**Comment by ichttt** [comment]
Whoops, you're right. I'll recheck everything and update the PR

File: src/main/java/net/minecraftforge/fluids/Fluid.java [hunk 2]
```diff
@@ -326,7 +326,7 @@ public String getLocalizedName(FluidStack stack)
     }
 
     /**
-     * A FluidStack sensitive version of getUnlocalizedName
+     * A FluidStack sensitive version of getTranslationKey
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/fluids/Fluid.java]: This should be kept as `getUnlocalizedName`, as it's referring to the method below.

File: src/main/java/net/minecraftforge/registries/IForgeRegistryEntry.java [hunk 3]
```diff
@@ -29,7 +29,7 @@
 {
     /**
      * Sets a unique name for this Item. This should be used for uniquely identify the instance of the Item.
-     * This is the valid replacement for the atrocious 'getUnlocalizedName().substring(6)' stuff that everyone does.
+     * This is the valid replacement for the atrocious 'getTranslationKey().substring(6)' stuff that everyone does.
      * Unlocalized names have NOTHING to do with unique identifiers. As demonstrated by vanilla blocks and items.
```
Comment (bs2609) [inline src/main/java/net/minecraftforge/registries/IForgeRegistryEntry.java]: Either leave this as-is, or change "Unlocalized names..." to "Translation keys..." as well.

**Comment by bs2609** [comment]
This needs an update to resolve conflicts.

**Comment by bs2609** [comment]
This was updated, but now it needs updating again, sorry.

**Comment by kierajreed** [comment]
This is still relevant, [the mappings haven't been changed](https://github.com/MinecraftForge/MinecraftForge/blob/1.12.x/build.gradle#L30).

**Pull request closed without merging**